src/core/gltf2importer/materialparser.cpp

```diff
@@ -149,24 +151,68 @@ Kuesa::MetallicRoughnessMaterial *createPbrMaterial(const Material &mat, const G
     return pbrMaterial;
 }
 
+Kuesa::UnlitMaterial *createUnlitMaterial(const Material &mat, const GLTF2Context *context)
+{
+    auto unlitMaterial = new Kuesa::UnlitMaterial();
+    unlitMaterial->setDoubleSided(mat.doubleSided);
+    unlitMaterial->setBaseColorFactor(QColor::fromRgbF(
+            mat.pbr.baseColorFactor[0],
+            mat.pbr.baseColorFactor[1],
+            mat.pbr.baseColorFactor[2],
+            mat.pbr.baseColorFactor[3]));
+
+    const qint32 baseColorTextureIdx = mat.pbr.baseColorTexture.index;
+    if (baseColorTextureIdx > -1) {
+        unlitMaterial->setBaseColorMap(context->texture(baseColorTextureIdx).texture);
+        unlitMaterial->setBaseColorUsesTexCoord1(mat.pbr.baseColorTexture.texCoord == 1);
+    }
+
+    switch (mat.alpha.mode) {
+    case Material::Alpha::Opaque:
+        unlitMaterial->setOpaque(true);
+        break;
+    case Material::Alpha::Blend:
+        unlitMaterial->setOpaque(false);
+        break;
+    case Material::Alpha::Mask:
+        unlitMaterial->setAlphaCutoffEnabled(true);
+        unlitMaterial->setAlphaCutoff(mat.alpha.alphaCutoff);
+    }
+
+    return unlitMaterial;
+}
+
 } // namespace
 
 Qt3DRender::QMaterial *Material::material(bool isSkinned, bool hasColorAttribute, const GLTF2Context *context)
 {
     if (isSkinned) {
         if (m_skinnedMaterial == nullptr) {
-            Kuesa::MetallicRoughnessMaterial *material = createPbrMaterial(*this, context);
-            material->setUseSkinning(true);
-            material->setUsingColorAttribute(hasColorAttribute);
-            m_skinnedMaterial = material;
+            if (extensions.KHR_materials_unlit) {
+                Kuesa::UnlitMaterial *material = createUnlitMaterial(*this, context);
+                material->setUseSkinning(true);
+                material->setUsingColorAttribute(hasColorAttribute);
+                m_skinnedMaterial = material;
```

Comment:
Later on it might make sense to introduce a GLTF2Material as a base class of metallicroughness and unlit, so that we can call setUseSkinning, setUsingColorAttributes for a single place.

Reply:
You are right. We can tackle this when we start supporting other material extensions